lib/search/query_components/highlight.rb

```diff
@@ -10,17 +10,21 @@ def payload
         fields: {
           title: {
             number_of_fragments: 0,
+            highlight_query: highlight_query(:title),
           },
           "title.synonym".to_sym => {
             number_of_fragments: 0,
+            highlight_query: highlight_query("title.synonym".to_sym),
```

Comment:
I _think_ the behaviour of highlighting synonyms is a benefit as it'll hopefully improve understanding of why a result has been returned.

It'd be interesting to sneak this into a benchmarking test